src/clients/SpokePoolClient.ts

```diff
@@ -710,17 +765,23 @@ export class SpokePoolClient extends BaseAbstractClient {
 
       const dataForQuoteTime = await this.batchComputeRealizedLpFeePct(depositEvents);
       for (const [index, event] of Array.from(depositEvents.entries())) {
-        const rawDeposit = spreadEventWithBlockNumber(event) as DepositWithBlock;
-
-        // Derive and append the destination token, LP fee, quote block number and block timestamp from the event.
-        // @dev Deposit events may _also_ include early deposits, in which case we did not retrieve a block.
-        const deposit: DepositWithBlock = {
-          ...rawDeposit,
-          realizedLpFeePct: dataForQuoteTime[index].realizedLpFeePct,
```

Comment:
Unfortunately I think we're still going to want to compute the realizedLpFeePct but we're going to want to associate them with the `Fill`. Or, we could leave this part in. This is to help aid the Dataworker, which needs to add an lpFee for each Fill that it is refunding, and probably also the relayer, which needs to compute the lpFee to determine profitability.

So I think we might want to keep this in here. This is something I'm realizing while scoping out the dataworker work.

We either need to compute realizedLpFeePct here, or in the dataworker step when constructing relayer refund roots

Reply:
Agree on keeping realizedLpFeePct - I might be a bit behind with this remote branch, but in the local state I have I am setting realizedLpFeePct unconditionally. There's actually no reason not to do it - we have the information available anyway.

Reply:
It's updated now. There's still a missing `relayerFeePct`. We have the option of computing it here, but that only works for like tokens, so I feel like it should be determined outside of the SpokePoolClient. It's only relevant to the relayer anyway, so it might slot into the ProfitClient instead (where we already have most of the tools to normalise to common prices anyway).

![diff](https://github.com/across-protocol/sdk-v2/assets/108695806/97c2c35f-385e-4954-8a3c-c4c68c71a88e)

Reply:
I agree its only relevant for the relayer. I think ProfitCLient makes sense!